netket/operator/_ising/base.py

```diff
@@ -184,18 +184,35 @@ def _iadd_same_hamiltonian(self, other):
             raise NotImplementedError(
                 "Cannot add hamiltonians on different hilbert spaces"
             )
+        if self._edges.shape[0] != other.edges.shape[0] or np.any(
+            [
+                edge not in np.sort(other._edges, axis=-1).tolist()
+                for edge in np.sort(self._edges, axis=-1).tolist()
+            ]
+        ):
```

Comment:
This is worse than quadratic complexity in the graph size (because you sort for every element in the edge) so it's not terribly ideal.

Then it's not an hot path, but in the past people complained of slow operations on operators so I would pay a tiny bit of attention.

I think we could simply do 
```suggestion
        if not np.allclose(self.edges, other.edges):
```
and call it a day. This will be fine 99% of times, always correct, and fast.

Reply:
Ok, I was trying to cover all the edge cases but since worst case the outcome is just a LocalOperator, I'm fine with this.